### ISSUE TITLE
[Tweak] Have LocomotionGrip apply throughout whole heirarchy

### ISSUE THREAD

**Issue by Hayden-da-Serg** [Frooxius/NeosPublic#647]
So while playing around with the LocomotionGrip component, I discovered that it does in fact disallow grip on the slots it's on no matter the world settings, however, in the case of automatically generated slots such as the ToolShelves, it's actually impossible to set the LocomotionGrip component to disallow gripping on them dynamically, I'd suggest that the LocomotionGrip applies throughout all of the child objects on the slot it's attached to if that object also has an ObjectRoot on it.

Use case being: I personally hate the fact that people can climb on me, and when I need my personal space, my forcefield does literally nothing when they're gripping onto an object, and I don't want to have to go to every collider on my avatar and attach a LocomotionGrip to them, and in the case of my ToolShelves, I can't make sure no one can climb them, I'd like to be able to apply the LocomotionGrip to the root of my avatar so no one is able to do so.

**Comment by Frooxius** [comment]
Hmm this is odd, it already applies to all children of the object it's applied to. Where exactly are you placing it in the hierarchy?

**Comment by H3BO3** [comment]
I think he's putting it on his avatar root from what I've heard

**Comment by Frooxius** [comment]
Where are the objects that users are still able to grip on relative to that?

**Comment by H3BO3** [comment]
The toolshelves, as he mentioned, which I assume are under the hand bones per usual. The toolshelf anchors have nothing in them by default but the shelves get generated upon avatar entry.

**Comment by Hayden-da-Serg** [comment]
Not the tool shelves specifically, just any collider on my avatar that doesn't have the component on it, and as @H3BO3 stated, yes it is on the root of the avatar and disabled, this seems to also only be the case if the locomotion is set to Grip on Any.
I personally have all of my anchors on an object under my avatar root with an AvatarPoseNode set to Controller Position because I cannot stand having limited anchors.

**Comment by H3BO3** [comment]
Since this is coming up again in the discord, it seems a deactivated LocomotionGrip does not prevent climbing on child colliders, only the collider it's alongside. Is this intended behavior?

**Comment by Cataena** [comment]
This would be useful to have fixed as a safety feature, as I've heard from quite a few people now who are uncomfortable with being climbed on.

**Comment by shiftyscales** [comment]
I've updated the title to be more reflective of the underlying issue (being unable to apply the component to the user root itself which is generated) rather than it being a failing of the component to apply to the entire hierarchy, because from the perspective of the component it does, and based on testimony, sounds like it does so properly?

Bronzewing has noted this issue can temporarily be addressed by using a tool he made to reparent the user's avatar and locomotion modules slot under a slot with the locomotion grip component present and disabled.

**Comment by ProbablePrime** [comment]
Cyro took a look at this, and wrote a mod which provides the option for this. It [adds the component to the root of the user](https://github.com/RileyGuy/HandsToYourself/blob/main/HandsToYourself.cs#L38).

Anything not parented to the root of the avatar, e.g Anything in the user root. Including the Locomotion colliders can still be grabbed with this component added to the avatar root. 

That's the issue. 

We'll need to update it, to look into the root of the avatar or change up the whole handling here somehow so it factors in avatar roots and user roots.

**Comment by LexiBasilisk** [comment]
Yeah- ideally in future we could make this a safety setting in which a user has to intentionally disable that safety to opt in to being grippable (regardless of world settings).

It seems like it'd hopefully be easy enough to add a setting for that (pending the settings UI rework as a pre-requisite of course.)
